src/jquery.valueview.Expert.js

```diff
@@ -120,12 +120,18 @@ jQuery.valueview = jQuery.valueview || {};
 
 		this._options = $.extend( ( !this._options ) ? {} : this._options, options || {} );
 
-		var defaultMessages = this._options.messages || {},
-			msgGetter = this._options.mediaWiki ? this._options.mediaWiki.msg : null;
-		this._messageProvider = new util.MessageProvider( {
-			defaultMessage: defaultMessages,
-			messageGetter: msgGetter
-		} );
+		if( this._options.messages ) {
+			this._messageProvider = new util.HashMessageProvider( this._options.messages );
+		}
+		if( this._options.messageProvider ) {
+			this._messageProvider = new util.CombiningMessageProvider(
+				this._options.messageProvider,
+				this._messageProvider
+			);
+		}
+		if( !this._messageProvider ) {
+			throw new Error( 'No message provider and no messages were provided to the valueview expert' );
```

Comment:
Documented. We have longer lines in this repository, I don't know what the upper limit is.